### PR TITLE
Feature/401516 sonar qube issues fix

### PR DESCRIPTION
**Pull request by Mazar-Shaik**

### CODE REVIEW

File: src/FrontendObligationChecker/Exceptions/BlobReaderException.cs [hunk 1]
```diff
@@ -20,8 +20,18 @@ public BlobReaderException(string message, Exception inner)
     {
     }
 
-    private BlobReaderException(SerializationInfo info, StreamingContext context)
+    protected BlobReaderException(SerializationInfo info, StreamingContext context)
         : base(info, context)
     {
     }
+
+    public override void GetObjectData(SerializationInfo info, StreamingContext context)
```
Comment (hardik-desai-kcl) [inline src/FrontendObligationChecker/Exceptions/BlobReaderException.cs:28]: Please add unit tests

File: src/FrontendObligationChecker/Exceptions/LargeProducerRegisterServiceException.cs [hunk 2]
```diff
@@ -20,8 +20,18 @@ public LargeProducerRegisterServiceException(string message, Exception inner)
     {
     }
 
-    private LargeProducerRegisterServiceException(SerializationInfo info, StreamingContext context)
+    protected LargeProducerRegisterServiceException(SerializationInfo info, StreamingContext context)
         : base(info, context)
     {
     }
+
+    public override void GetObjectData(SerializationInfo info, StreamingContext context)
```
Comment (hardik-desai-kcl) [inline src/FrontendObligationChecker/Exceptions/LargeProducerRegisterServiceException.cs:28]: please add unit tests

File: src/FrontendObligationChecker/Services/PageService/PageService.cs [hunk 3]
```diff
@@ -162,14 +162,19 @@ private async Task SetAssociationTypeAsync(Page page)
         var sessionPage = sessionJourney?.Pages.SingleOrDefault(x => x.Path == PagePath.TypeOfOrganisation);
         if (sessionPage != null)
         {
-            var answer = sessionPage.Questions.FirstOrDefault(q => q.Key == QuestionKey.TypeOfOrganisation).Answer;
-            page.AssociationType = answer switch
+            var question = sessionPage.Questions.Find(q => q.Key == QuestionKey.TypeOfOrganisation);
```
Comment (hardik-desai-kcl) [inline src/FrontendObligationChecker/Services/PageService/PageService.cs:165]: Please add unit test as I think this is change in behaviour

File: src/FrontendObligationChecker/ISharedResources.cs [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ﻿namespace FrontendObligationChecker;
 
-public class SharedResources
+public interface ISharedResources
```
Comment (hardik-desai-kcl) [inline src/FrontendObligationChecker/ISharedResources.cs:3]: can we not just delete this empty class?

File: src/FrontendObligationChecker/ViewModels/AmountYouSupply.cs [hunk 5]
```diff
@@ -40,30 +40,32 @@ private static List<Question> GetNonSellerActivityYesQuestions(Page page)
...
+              && page.FindPage(PagePath.MaterialsForDrinksContainers)?.Questions.FirstOrDefault()?.Answer != "0";
 
         if (!isUsingDrinkContainersMaterials)
         {
             return false;
         }
 
-        var containerVolumePage = page.FindPage(PagePath.ContainerVolume);
-        return containerVolumePage?.Questions.FirstOrDefault().Answer == YesNo.Yes;
+        return page.FindPage(PagePath.ContainerVolume)?.Questions.FirstOrDefault()?.Answer != null
```
Comment (hardik-desai-kcl) [inline src/FrontendObligationChecker/ViewModels/AmountYouSupply.cs:60]: Please add a unit test, not sure why you are calling 
page.FindPage(PagePath.ContainerVolume) and Questions.FirstOrDefault()?.Answer twice

File: src/FrontendObligationChecker/Views/ObligationChecker/WhatYouNeedToDo.cshtml [hunk 6]
```diff
@@ -140,7 +140,7 @@
                     }
                     else
                     {
-                        @SharedLocalizer[firstPage.TitleCaption]
+                        @SharedLocalizer[firstPage?.TitleCaption]
```
Comment (hardik-desai-kcl) [inline src/FrontendObligationChecker/Views/ObligationChecker/WhatYouNeedToDo.cshtml:143]: please add a unit test

File: src/FrontendObligationChecker/Views/ObligationChecker/WhatYouNeedToDo.cshtml [hunk 7]
```diff
@@ -150,19 +150,22 @@
...
+                            var noneOption = currentPage.FirstQuestion?.Options.Find(o => o.IsSelected == true && o.Value == "0");
                             if (noneOption != null)
                             {
                                 <p class="govuk-body">@SharedLocalizer[noneOption.Title]</p>
                             }
                             else
                             {
                                 <ul class="govuk-list govuk-list--bullet">
-                                    @foreach (var option in currentPage.FirstQuestion?.Options.Where(o => o.IsSelected == true))
+                                    @if (currentPage.FirstQuestion?.Options != null)
```
Comment (hardik-desai-kcl) [inline src/FrontendObligationChecker/Views/ObligationChecker/WhatYouNeedToDo.cshtml:161]: please add unit test

**Review by hardik-desai-kcl — CHANGES_REQUESTED** [review]

**Review by davidtonthat — CHANGES_REQUESTED** [review]

File: src/FrontendObligationChecker/Program.cs [hunk 8]
```diff
@@ -1,81 +1,85 @@
...
-app.UseRequestLocalization();
-app.UseMiddleware<AnalyticsCookieMiddleware>();
-app.MapHealthChecks(builder.Configuration.GetValue<string>("HEALTH_CHECK_LIVENESS_PATH"), HealthCheckOptionBuilder.Build());
-app.MapControllers();
-app.Run();
-
-public partial class Program
+namespace FrontendObligationChecker
 {
+    public partial class Program
```
[the remark this answers is not in the document]
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/Program.cs:9]: @davidtonthat I agree with as per new C# coding standard we don't need enclosing but SQ throwing an error if its having
public partial class program
{
}  
at the end. I can't remove this as its using in few other places. Only way to get rid of this SQ issue is enclosing everything under braces. Hope it make sense.

File: src/FrontendObligationChecker/Program.cs [hunk 9]
```diff
@@ -1,81 +1,85 @@
-﻿using FrontendObligationChecker;
-using FrontendObligationChecker.ConfigurationExtensions;
+﻿using FrontendObligationChecker.ConfigurationExtensions;
 using FrontendObligationChecker.HealthChecks;
 using FrontendObligationChecker.Middleware;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.FeatureManagement;
 
-var builder = WebApplication.CreateBuilder(args);
```
[the remark this answers is not in the document]
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/Program.cs:8]: @davidtonthat I agree with as per new C# coding standard we don't need enclosing but SQ throwing an error if its having
public partial class program
{
}
at the end. I can't remove this as its using in few other places. Only way to get rid of this SQ issue is enclosing everything under braces. Hope it make sense.

File: src/FrontendObligationChecker/Exceptions/BlobReaderException.cs (hunk 1 above)
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/Exceptions/BlobReaderException.cs]: Removed this as it is no need.

File: src/FrontendObligationChecker/Exceptions/LargeProducerRegisterServiceException.cs (hunk 2 above)
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/Exceptions/LargeProducerRegisterServiceException.cs]: Removed this as it is no need.

File: src/FrontendObligationChecker/Services/PageService/PageService.cs (hunk 3 above)
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/Services/PageService/PageService.cs]: updated

File: src/FrontendObligationChecker/ISharedResources.cs (hunk 4 above)
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/ISharedResources.cs]: updated

File: src/FrontendObligationChecker/Views/ObligationChecker/WhatYouNeedToDo.cshtml (hunk 7 above)
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/Views/ObligationChecker/WhatYouNeedToDo.cshtml]: updated

File: src/FrontendObligationChecker/Views/ObligationChecker/WhatYouNeedToDo.cshtml (hunk 6 above)
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/Views/ObligationChecker/WhatYouNeedToDo.cshtml:143]: ignored as discussed

File: src/FrontendObligationChecker/ViewModels/AmountYouSupply.cs [hunk 10]
```diff
@@ -39,31 +39,31 @@ private static List<Question> GetNonSellerActivityYesQuestions(Page page)
 
     private static bool CheckHasSingleUseDrinkContainers(Page page)
     {
-        var isPlacingDrinksOnMarket =
-            page.FindPage(PagePath.PlaceDrinksOnMarket)?.Questions.FirstOrDefault().Answer == YesNo.Yes;
+        var placingDrinksOnMarket = page.FindPage(PagePath.PlaceDrinksOnMarket)?.Questions?.FirstOrDefault()?.Answer;
+        var isPlacingDrinksOnMarket = placingDrinksOnMarket == YesNo.Yes;
```
[the remark this answers is not in the document]
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/ViewModels/AmountYouSupply.cs:43]: To fix SonarQube 10.4 issue "is null on at least one execution path" on this line.

File: src/FrontendObligationChecker/ViewModels/AmountYouSupply.cs [hunk 11]
```diff
@@ -39,31 +39,31 @@ private static List<Question> GetNonSellerActivityYesQuestions(Page page)
...
 
         if (!isPlacingDrinksOnMarket)
         {
             return false;
         }
 
-        bool isUsingDrinkContainersMaterials =
-            page.FindPage(PagePath.MaterialsForDrinksContainers)?.Questions.FirstOrDefault().Answer != "0";
+        var drinkContainersMaterials = page.FindPage(PagePath.MaterialsForDrinksContainers)?.Questions?.FirstOrDefault()?.Answer;
```
[the remark this answers is not in the document]
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/ViewModels/AmountYouSupply.cs:50]: To fix SonarQube 10.4 issue "is null on at least one execution path" on this line.

File: src/FrontendObligationChecker/ViewModels/AmountYouSupply.cs [hunk 12]
```diff
@@ -39,31 +39,31 @@ private static List<Question> GetNonSellerActivityYesQuestions(Page page)
...
+        bool isUsingDrinkContainersMaterials = drinkContainersMaterials != "0";
 
         if (!isUsingDrinkContainersMaterials)
         {
             return false;
         }
 
-        var containerVolumePage = page.FindPage(PagePath.ContainerVolume);
-        return containerVolumePage?.Questions.FirstOrDefault().Answer == YesNo.Yes;
+        var answer = page.FindPage(PagePath.ContainerVolume)?.Questions?.FirstOrDefault()?.Answer;
```
[the remark this answers is not in the document]
Comment (Mazar-Shaik) [reply src/FrontendObligationChecker/ViewModels/AmountYouSupply.cs:58]: To fix SonarQube 10.4 issue "is null on at least one execution path" on this line.

**Review by hardik-desai-kcl — APPROVED** [review]

**Review by davidtonthat — APPROVED** [review]

**Review by hardik-desai-kcl — DISMISSED** [review]

**Review by hardik-desai-kcl — APPROVED** [review]

File: src/FrontendObligationChecker.UnitTests/Services/NextFinder/PageFinderTests/AmountYouSupplyTests.cs [hunk 13]
```diff
@@ -1,23 +1,47 @@
...
+    private static readonly int[] TestScenario15 =[1, 1, 0, 0, 1, 1, 1];
+    private static readonly int[] TestScenario16 =[0, 0, 1, 1, 0, 0, 0];
+    private static readonly int[] TestScenario17 =[0, 0, 0, 0, 1, 1, 1];
+    private static readonly int[] TestScenario18 =[1, 1, 1, 0, 0, 0, 0];
+    private static readonly string[] TestScenario19 =["Under 25 tonnes", "1"];
+    private static readonly string[] TestScenario20 =["25 tonnes to 50 tonnes", "2"];
+    private static readonly string[] TestScenario21 =["More than 50 tonnes", "3"];
+
     [TestMethod]
-    [DataRow(new[] { 0, 0, 1, 0, 0, 0, 0 })]
```
Comment (davidtonthat) [inline src/FrontendObligationChecker.UnitTests/Services/NextFinder/PageFinderTests/AmountYouSupplyTests.cs:10]: What issues were there with this code? It looks fine to me!

**Review by davidtonthat — APPROVED** [review]

**Review by EvidenPete — APPROVED** [review]

**Review by hardik-desai-kcl — APPROVED** [review]

**Pull request merged**